core/templates/dev/head/pages/learner_dashboard/LearnerDashboard.js

```diff
@@ -0,0 +1,262 @@
+      $scope.activeSection = newActiveSectionName;
+    };
+
+    $scope.setActiveSubSection = function(newActiveSubSectionName) {
+      $scope.activeSubSection = newActiveSubSectionName;
+    };
+
+    $scope.getExplorationUrl = function(explorationId) {
+      return '/explore/' + explorationId;
+    };
+
+    $scope.getCollectionUrl = function(collectionId) {
+      return '/collection/' + collectionId;
+    };
+
+    $scope.checkMobileView = function() {
+      return ($window.innerWidth < 500);
+    };
+
+    $scope.showUsernamePopover = function(subscriberUsername) {
+      // The popover on the subscription card is only shown if the length of
+      // the subscriber username is greater than 10 and the user hovers over
+      // the truncated username.
+      if (subscriberUsername.length > 10) {
+        return 'mouseenter';
+      } else {
+        return 'none';
+      }
+    };
+
+    $scope.setExplorationsSortingOptions = function(sortType) {
+      if (sortType === $scope.currentExpSortType) {
+        $scope.isCurrentExpSortDescending = !$scope.isCurrentExpSortDescending;
+      } else {
+        $scope.currentExpSortType = sortType;
+      }
+      console.log($scope.currentExpSortType);
+    };
+
+    $scope.setSubscriptionSortingOptions = function(sortType) {
+      if (sortType === $scope.currentSubscriptionSortType) {
+        $scope.isCurrentSubscriptionSortDescending = (
+          !$scope.isCurrentSubscriptionSortDescending);
+      } else {
+        $scope.currentSubscriptionSortType = sortType;
+      }
+    };
+
+    $scope.sortSubscriptionFunction = function(entity) {
+      // This function is passed as a custom comparator function to `orderBy`,
+      // so that special cases can be handled while sorting subscriptions.
+      var value = entity[$scope.currentSubscribersSortType];
+      if ($scope.currentSubscribersSortType ===
+          SUBSCRIPTION_SORT_BY_KEYS.IMPACT) {
+        value = (value || 0);
+      }
+      return value;
+    };
+
+    $scope.openRemoveEntityModal = function(
+      sectionName, subSectionName, entity) {
+      $modal.open({
+        templateUrl: 'modals/removeEntity',
+        backdrop: true,
+        resolve: {
+          sectionName: function() {
+            return sectionName;
+          },
+          subSectionName: function() {
+            return subSectionName;
+          },
+          entity: function() {
+            return entity;
+          }
+        },
+        controller: [
+          '$scope', '$modalInstance', '$http', 'sectionName', 'subSectionName',
+          function($scope, $modalInstance, $http, sectionName, subSectionName) {
+            $scope.sectionName = sectionName;
+            $scope.subSectionName = subSectionName;
+            $scope.entityTitle = entity.title;
+
+            $scope.remove = function() {
+              /* eslint-disable max-len */
+              if (subSectionName === LEARNER_DASHBOARD_SUB_SECTIONS.EXPLORATIONS) {
```

Comment:
Argh, yes, sorry. I don't have a better suggestion...

The only thing I would suggest, though, that is somewhat unrelated to this, is to use SUBSECTION instead of SUB_SECTION, and subsection instead of subSection. "subsection" is one word.